ttnn/cpp/ttnn/operations/sharding_utilities.hpp

```diff
@@ -91,15 +92,39 @@ ShardingConfig get_specs_for_sharding_partition(
 
 namespace sharded_accessor_utils {
 
+enum class ArgConfig : uint8_t {
+    CTA = 0,
+    RankCRTA = 1 << 0,
+    NumBanksCRTA = 1 << 1,
+    TensorShapeCRTA = 1 << 2,
+    ShardShapeCRTA = 1 << 3,
+    BankCoordsCRTA = 1 << 4,
+    CRTA = RankCRTA | NumBanksCRTA | TensorShapeCRTA | ShardShapeCRTA | BankCoordsCRTA
+};
+
+using ArgsConfig = Flags<ArgConfig>;
+constexpr ArgsConfig operator|(ArgConfig a, ArgConfig b) noexcept { return ArgsConfig(a) | b; }
+constexpr ArgsConfig operator|(ArgConfig a, ArgsConfig b) noexcept { return ArgsConfig(a) | b; }
+
 struct ShardedAccessorArgs {
-    size_t rank;
-    size_t num_banks;
-    std::vector<uint32_t> shapes_and_bank_coords;
+    uint32_t get_rank() const;
+    uint32_t get_num_banks() const;
+    uint32_t get_physical_num_banks() const;
+    tt::stl::Span<const uint32_t> get_tensor_shape() const;
+    tt::stl::Span<const uint32_t> get_shard_shape() const;
+    tt::stl::Span<const uint32_t> get_bank_coords() const;
+
```

Comment:
Add these would much more reduce code footprint in get_rank, get_num_banks, etc.
```suggestion
    bool rank_is_crta() const { return args_config.contains(ArgConfig::RankCRTA); }
    bool banks_is_crta() const { return args_config.contains(ArgConfig::NumBanksCRTA); }
    bool tensor_is_crta() const { return args_config.contains(ArgConfig::TensorShapeCRTA); }
    bool shard_is_crta() const { return args_config.contains(ArgConfig::ShardShapeCRTA); }
    bool bank_coords_is_crta() const { return args_config.contains(ArgConfig::BankCoordsCRTA); }
```

Reply:
I just left some suggestion for your considerations.

Reply:
Okay you convinced me :) I will commit your suggestions

Reply:
2 files changed, 36 insertions(+), 55 deletions(-)